### ISSUE TITLE
Investigate PPO opt state dict checkpointing

### ISSUE THREAD

**Issue by SalmanMohammadi** [pytorch/torchtune#2565]

**Comment by SalmanMohammadi** [comment]
> Any strong opposition to deprecating optim in backward?

This single device recipe in particular heavily relies on the memory savings form opt in bwd.

**Comment by pbontrager** [comment]
I realize that every MB counts when so much is being crammed on one GPU, but it does mean you can't do gradient accumulation forcing you to very small batch sizes. What is the memory requirement difference between opt_in_bwd=True/False?
